File: src/game/tools/rpc/output/wrap/ItemDataListWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class ItemDataListWrap : IWrapper
	{	
		public ItemDataWrap[] ItemList; // 道具列表
		
		public void Decode(Packet pck)
		{	
			ItemList = new ItemDataWrap[pck.GetShort()];
			for (int i = 0; i < ItemList.Length; i++)
			{
				ItemList[i] = new ItemDataWrap();
				ItemList[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (ItemList == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)ItemList.Length);
	        	for(int i = 0; i < ItemList.Length; i++)
	        	{
	        		ItemList[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}

File: src/game/tools/rpc/output/wrap/ItemDataListWrap.ts
class ItemDataListWrap extends Net.Simple.IWrapper {
	
	public ItemList:Array<ItemDataWrap>; // 道具列表
	public constructor() {
		super();
	}
	Decode(pck:Net.Simple.Packet):ItemDataListWrap{
		
		this.ItemList = new Array<ItemDataWrap>();
		for (var i = 0,len = this.ItemList.length; i < len; i++)
		{
			this.ItemList[i] = new ItemDataWrap();
			this.ItemList[i].Decode(pck);
			
		}
		
		return this;
	}
    Encode(pck:Net.Simple.Packet){
		
    	if (this.ItemList == null) pck.PutShort(0); 
    	else
    	{
        	pck.PutShort(this.ItemList.length);
        	for(var i = 0,len = this.ItemList.length; i < len; i++)
        	{
        		this.ItemList[i].Encode(pck);
				
        	}
        }
    	
	}

}